src/components/ble/weather/WeatherService.cpp

```diff
@@ -552,11 +552,12 @@ namespace Pinetime {
 
     int16_t WeatherService::GetTodayMinTemp() const {
       uint64_t currentTimestamp = GetCurrentUnixTimestamp();
-      uint64_t currentDayEnd = currentTimestamp - ((24 - dateTimeController.Hours()) * 60 * 60) -
-                               ((60 - dateTimeController.Minutes()) * 60) - (60 - dateTimeController.Seconds());
+      uint64_t currentDayEnd = currentTimestamp + ((24 - dateTimeController.Hours()) * 60 * 60) +
+                               ((60 - dateTimeController.Minutes()) * 60) + (60 - dateTimeController.Seconds());
+      uint64_t currentDayStart = currentDayEnd - 86400;
       int16_t result = -32768;
       for (auto&& header : this->timeline) {
-        if (header->eventType == WeatherData::eventtype::Temperature && IsEventStillValid(header, currentTimestamp) &&
+        if (header->eventType == WeatherData::eventtype::Temperature && header->timestamp > currentDayStart &&
```

Comment:
It should either be `header->timestamp >= currentDayStart && header->timestamp < currentDayEnd` or `header->timestamp > currentDayStart && header->timestamp <= currentDayEnd` otherwise when `header->timestamp == currentDayStart` or `header->timestamp == nextDayStart` events will be ignored.

Reply:
Thanks